test/testPartyAttachments.js

```diff
@@ -0,0 +1,1025 @@
+            },
+            resource: {
+              href: resourceCopyHref,
+            },
+          },
+        ];
+
+        const copyAttResult = await httpClient.post({
+          path: `${type}/attachments/copy`,
+          body: copyAttBody,
+        });
+        assert.equal(copyAttResult.status, 200);
+
+        // verify if the copied attachments are present
+        const attachmentCopyUrl1 =
+          type + "/" + resourceCopyKey + "/attachments/profile1.png";
+        const attachmentCopyUrl2 =
+          type + "/" + resourceCopyKey + "/attachments/profile2.png";
+
+        const responseGetCopyAtt1 = await httpClient.get({ path: attachmentCopyUrl1 });
+        assert.equal(responseGetCopyAtt1.status, 200);
+
+        const responseGetCopyAtt2 = await httpClient.get({ path: attachmentCopyUrl2 });
+        assert.equal(responseGetCopyAtt2.status, 200);
+      });
+
+      it("copy attachments without fileHref should result in proper error", async () => {
+        const attachmentKey1 = uuid.v4();
+        const resourceCopyKey = uuid.v4();
+        const resourceCopyHref = type + "/" + resourceCopyKey;
+        const attachmentCopyKey1 = uuid.v4();
+
+        const copyAttBody = [
+          {
+            file: "profile1.png",
+            attachment: {
+              key: attachmentCopyKey1,
+              description: `this is MY file with key ${attachmentKey1}`,
+            },
+            resource: {
+              href: resourceCopyHref,
+            },
+          },
+        ];
+        const copyAttResult = await httpClient.post({
+          path: `${type}/attachments/copy`,
+          body: copyAttBody,
+        });
+        console.log(copyAttResult.status);
+        console.log(copyAttResult.body);
+
+        assert.equal(copyAttResult.status, 400);
+        assert.equal(
+          copyAttResult.body.errors[0].code,
+          "missing.json.fileHref"
+        );
+      });
+
+      it("missing body should result in proper error", async () => {
+        const copyAttResult = await httpClient.post({
+          path: `${type}/attachments/copy`,
+          body: undefined,
+        });
+
+        console.log(copyAttResult.status);
+        console.log(copyAttResult.body);
+
+        assert.equal(copyAttResult.status, 400);
+        assert.equal(
+          copyAttResult.body.errors[0].code,
+          "missing.json.body.attachment"
+        );
+      });
+    });
+  });
+};
+
+
+// TODO: test rollback (delete of temp files) in case of error
+// TODO: check if mimetype is set as expected
+
+// No testcase for customRouteForPreSignedUpload provided yet as it is not used
+// TODO : test security??
+
+// TODO : Define resource with S3 and file storage to test both
+// TODO : When BLOB database storage is implemented, also add a resource on that with tests
+// TODO : Implement + check after & before function (with database access) on GET, PUT and DELETE.
```

Comment:
TODO: add test for filename with special characters (check originalFilename and filename).